www/base/src/app/builders/builders.controller.coffee

```diff
@@ -102,12 +102,28 @@ class Builders extends Controller
 
         data = dataService.open().closeOnDestroy($scope)
 
-        $scope.builders = data.getBuilders()
-        $scope.masters = data.getMasters()
-
         # as there is usually lots of builders, its better to get the overall
         # list of workers, masters, and builds and then associate by builder
+        $scope.builders = data.getBuilders()
+        $scope.masters = data.getMasters()
         workers = data.getWorkers()
-        builds = data.getBuilds(limit: 200, order: '-started_at')
-        dataGrouperService.groupBy($scope.builders, workers, 'builderid', 'workers', 'configured_on')
-        dataGrouperService.groupBy($scope.builders, builds, 'builderid', 'builds')
+        builds = null
+
+        requeryBuilds = () ->
+            $scope.builders.forEach (builder) -> builder.builds = []
+
+            filteredBuilds = $filter('filter')($scope.builders, $scope.isBuilderFiltered) || []
+            builderIds = filteredBuilds.map (builder) -> builder.builderid
+            builderIds = [] if builderIds.length == $scope.builders.length
+
+            builds = data.getBuilds(limit: 200, order: '-started_at', builderid__eq: builderIds)
+            dataGrouperService.groupBy($scope.builders, workers, 'builderid', 'workers', 'configured_on')
+            dataGrouperService.groupBy($scope.builders, builds, 'builderid', 'builds')
+
+        $scope.$watch "builders.$resolved", (resolved) -> requeryBuilds() if resolved
```

Comment:
This adds a delay on the list of builds. We should avoid that at least when no tag is selected

Reply:
done for the case when no tag is selected